src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/dft/dft_kernel_ref.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2021-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "dft_kernel_ref.h"
+
+#include <kernel_selector_utils.h>
+#include <iostream>
+namespace kernel_selector {
+
+namespace {
+
+CommonDispatchData SetDefault(const dft_params &params) {
+    CommonDispatchData dispatchData;
+
+    auto outDims = params.output.LogicalDims();
+    // opencl kernels have inverted order of dimensions with respect to axis spec: x is smallest index, b is largest
+    // we are skipping x, since it contains complex pairs
+    auto complexSize = std::accumulate(outDims.begin() + 1, outDims.end(), size_t{1}, std::multiplies<size_t>{});
+    dispatchData.gws = { 1, 1, complexSize }; // TODO: these could be split better
+    dispatchData.lws = GetOptimalLocalWorkGroupSizes(dispatchData.gws, params.engineInfo);
+
+    return dispatchData;
+}
+
+}  // namespace
+
+KernelsData DFTKernelRef::GetKernelsData(const Params &params, const optional_params &options) const {
+    KernelsData kernels_data;
+    if (!Validate(params, options))
+        return kernels_data;
+    kernels_data.push_back(KernelData::Default<dft_params>(params));
+    KernelData &kernel_data = kernels_data.front();
+    auto &derived_params = dynamic_cast<dft_params&>(*kernel_data.params.get());
+    auto dispatch_data = SetDefault(derived_params);
+    auto entry_point = GetEntryPoint(kernelName, derived_params.layerID, params, options);
+    auto jit_constants = GetJitConstants(derived_params);
+    auto jit = CreateJit(kernelName, jit_constants, entry_point);
+    auto &clKernelData = kernel_data.kernels[0];
+    FillCLKernelData(clKernelData, dispatch_data, params.engineInfo, kernelName, jit, entry_point);
+    return kernels_data;
+}
+
+KernelsPriority DFTKernelRef::GetKernelsPriority(const Params& /*params*/, const optional_params& /*options*/) const {
+    return DONT_USE_IF_HAVE_SOMETHING_ELSE;
+}
+
+ParamsKey DFTKernelRef::GetSupportedKey() const {
+    ParamsKey k;
+    k.EnableInputDataType(Datatype::F16);
+    k.EnableInputDataType(Datatype::F32);
+    k.EnableOutputDataType(Datatype::F16);
+    k.EnableOutputDataType(Datatype::F32);
+    k.EnableInputLayout(DataLayout::bfzyx);
+    k.EnableOutputLayout(DataLayout::bfzyx);
```

Comment:
Why does it only support bfzyx?

Reply:
Added other dimensions.

Reply:
Could you please add those layout for input/output? Those are required for dGPU support.
<html>
<body>
<!--StartFragment-->

Prec | Layout
-- | --
int8 | fsv16, fsv32, bsv32_fsv32
fp16 | fsv16, bsv32_fsv16
fp32 | fsv16, bsv32_fsv16

<!--EndFragment-->
</body>
</html>

Reply:
agreed not to do for now